src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/ConnectionTest.php

```diff
@@ -271,57 +287,90 @@ public function testClaimAbandonedMessageWithRaceCondition()
     {
         $redis = $this->createMock(\Redis::class);
 
-        $redis->expects($this->exactly(3))->method('xreadgroup')
-            ->willReturnCallback(function (...$args) {
-                static $series = [
-                    // first call for pending messages
-                    [['symfony', 'consumer', ['queue' => '0'], 1, null], []],
-                    // second call because of claimed message (redisid-123)
-                    [['symfony', 'consumer', ['queue' => '0'], 1, null], []],
-                    // third call because of no result (other consumer claimed message redisid-123)
-                    [['symfony', 'consumer', ['queue' => '>'], 1, null], []],
-                ];
-
-                [$expectedArgs, $return] = array_shift($series);
-                $this->assertSame($expectedArgs, $args);
-
-                return $return;
-            })
-        ;
+        // xpending -> xclaim -> xpending -> xreadgroup
+        $redis->expects($this->once())->method('xreadgroup')
+            ->with('symfony', 'consumer', ['queue' => '>'], 1, null) // weren't able to claim the message, asking for new ones
+            ->willReturn([]);
 
-        $redis->expects($this->once())->method('xpending')->willReturn([[
-            0 => 'redisid-123', // message-id
-            1 => 'consumer-2', // consumer-name
-            2 => 3600001, // idle
-        ]]);
+        $redis->expects($this->exactly(2))->method('xpending')->willReturnOnConsecutiveCalls(
+            [[
+                0 => 'redisid-123', // message-id
+                1 => 'consumer-2', // consumer-name
+                2 => 3600001, // idle
+            ]],
+            []
+        );
 
         $redis->expects($this->exactly(1))->method('xclaim')
             ->with('queue', 'symfony', 'consumer', 3600000, ['redisid-123'], ['JUSTID'])
+            ->willReturn([]);   // other consumer claimed the message first
+
+        $connection = Connection::fromDsn('redis://localhost/queue', ['delete_after_ack' => true], $redis);
+        $connection->get();
+    }
+
+    public function testClaimAbandonedMessageWithRaceConditionExtremeContention()
+    {
+        $redis = $this->createMock(\Redis::class);
+
+        // (xpending -> xclaim) x 3 -> xreadgroup
+        $redis->expects($this->once())->method('xreadgroup')
+            ->with('symfony', 'consumer', ['queue' => '>'], 1, null) // weren't able to claim a message, asking for new ones
             ->willReturn([]);
 
+        $redis->expects($this->exactly(3))->method('xpending')->willReturnOnConsecutiveCalls(
```

Comment:
here also, willReturnOnConsecutiveCalls should be replaced, PHPUnit deprecated it.